examples/tutorials/nb_python/habitat_lab.py

```diff
@@ -0,0 +1,398 @@
+
+import math
+import os
+import random
+import sys
+
+import magnum as mn
+import numpy as np
+from gym import spaces
+
+# %matplotlib inline
+from matplotlib import pyplot as plt
+
+import habitat_sim
+from habitat_sim.utils import common as utils
+from habitat_sim.utils import viz_utils as vut
+
+# %cd "/content/habitat-api"
+
+
+if "google.colab" in sys.modules:
+    # This tells imageio to use the system FFMPEG that has hardware acceleration.
+    os.environ["IMAGEIO_FFMPEG_EXE"] = "/usr/bin/ffmpeg"
+
+
+from PIL import Image
+
+import habitat
+import habitat_baselines
+from habitat.core.embodied_task import EmbodiedTask
+from habitat.core.logging import logger
+from habitat.core.registry import registry
+from habitat.tasks.nav.nav import NavigationTask
+from habitat_baselines.common.baseline_registry import baseline_registry
+from habitat_baselines.config.default import get_config as get_baselines_config
+
+# %%
+# @title Define Observation Display Utility Function { display-mode: "form" }
+
+# @markdown A convenient function that displays sensor observations with matplotlib.
+
+# @markdown (double click to see the code)
+
+
+# Change to do something like this maybe: https://stackoverflow.com/a/41432704
+def display_sample(rgb_obs, semantic_obs=np.array([]), depth_obs=np.array([])):
+    from habitat_sim.utils.common import d3_40_colors_rgb
+
+    rgb_img = Image.fromarray(rgb_obs, mode="RGB")
+
+    arr = [rgb_img]
+    titles = ["rgb"]
+    if semantic_obs.size != 0:
+        semantic_img = Image.new(
+            "P", (semantic_obs.shape[1], semantic_obs.shape[0])
+        )
+        semantic_img.putpalette(d3_40_colors_rgb.flatten())
+        semantic_img.putdata((semantic_obs.flatten() % 40).astype(np.uint8))
+        semantic_img = semantic_img.convert("RGBA")
+        arr.append(semantic_img)
+        titles.append("semantic")
+
+    if depth_obs.size != 0:
+        depth_img = Image.fromarray(
+            (depth_obs / 10 * 255).astype(np.uint8), mode="L"
+        )
+        arr.append(depth_img)
+        titles.append("depth")
+
+    plt.figure(figsize=(12, 8))
+    for i, data in enumerate(arr):
+        ax = plt.subplot(1, 3, i + 1)
+        ax.axis("off")
+        ax.set_title(titles[i])
+        plt.imshow(data)
+    plt.show(block=False)
+
+
+# %% [markdown]
+# ## Setup PointNav Task
+
+# %%
+# cat "/content/habitat-api/configs/test/habitat_all_sensors_test.yaml"
+
+# %%
+config = habitat.get_config(
+    config_paths="./configs/test/habitat_all_sensors_test.yaml"
+)
+
+env = habitat.Env(config=config)
+
+# %%
+action = None
+obs = env.reset()
+valid_actions = ["TURN_LEFT", "TURN_RIGHT", "MOVE_FORWARD", "STOP"]
+
+while action != "STOP":
+    display_sample(obs["rgb"])
+    print(
+        "distance to goal: {:.2f}".format(obs["pointgoal_with_gps_compass"][0])
+    )
+    print(
+        "angle to goal (radians): {:.2f}".format(
+            obs["pointgoal_with_gps_compass"][1]
+        )
+    )
+    action = input(
```

Comment:
@abhiskk Taking input like this makes it impossible to run the script. We need to either a script flag to pick an action randomly to test the script or otherwise modify how the actions are chosen by the user.